docs/running-on-yarn.md

```diff
@@ -30,6 +30,22 @@ Most of the configs are the same for Spark on YARN as for other deployment modes
   </td>
 </tr>
 <tr>
+  <td><code>spark.driver.cores</code></td>
+  <td>1</td>
+  <td>
+    Number of cores to use for the YARN Application Master in cluster mode.
```

Comment:
it's weird how the description for `spark.driver.cores` only talks about the YARN application master. Maybe this should be more like the following:

Number of cores used by the driver in YARN cluster mode. Since the driver is run in the same JVM as the YARN Application Master in cluster mode, this also controls the cores used by the YARN AM. In client mode, use ... instead.